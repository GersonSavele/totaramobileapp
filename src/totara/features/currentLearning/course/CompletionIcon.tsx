/**
 * This file is part of Totara Enterprise.
 *
 * Copyright (C) 2020 onwards Totara Learning Solutions LTD
 *
 * Totara Enterprise is provided only to Totara Learning Solutions
 * LTDâ€™s customers and partners, pursuant to the terms and
 * conditions of a separate agreement with Totara Learning
 * Solutions LTD or its affiliate.
 *
 * If you do not have an agreement with Totara Learning Solutions
 * LTD, you may not access, use, modify, or distribute this software.
 * Please contact [sales@totaralearning.com] for more information.
 */

import React from "react";
import { StyleSheet, TouchableOpacity } from "react-native";
import { CircleIcon } from "@totara/components";
import { iconSizes, margins } from "@totara/theme/constants";
import { Spinner } from "native-base";
import { getCompletionStatus } from "../utils";

type BuildContentProps = {
  completion?: string;
  status?: string;
  available: boolean;
  loading?: boolean;
  onPress?: () => void;
  onGetCompletionStatus?: Function;
};

const CompletionIcon = ({
  completion,
  status,
  available,
  loading,
  onPress,
  onGetCompletionStatus = getCompletionStatus
}: BuildContentProps) => {
  const isDisabled = loading || !onPress;
  const completionStatusData = onGetCompletionStatus({
    available,
    completion,
    status
  });
  if (!completionStatusData) {
    return null;
  } else {
    const { stateObj, accessibility } = completionStatusData;

    const accessibleLable = accessibility?.label;
    const accessibleRole = accessibility?.role;
    const accessibleState = loading ? { busy: true } : accessibility?.state;

    return (
      <TouchableOpacity
        onPress={onPress}
        style={{ marginRight: margins.marginL }}
        disabled={isDisabled}
        accessibilityLabel={accessibleLable}
        accessibilityRole={accessibleRole}
        accessibilityState={accessibleState}
        accessible={true}>
        {loading ? (
          <Spinner size={iconSizes.sizeM} style={styles.spinner} />
        ) : (
          <CircleIcon
            icon={stateObj.icon}
            backgroundColor={stateObj.backgroundColor}
            iconColor={stateObj.iconColor}
            borderColor={stateObj.borderColor}
            fontAwesomeIcon={stateObj.fontAwesomeIcon}
          />
        )}
      </TouchableOpacity>
    );
  }
};

const styles = StyleSheet.create({
  spinner: {
    height: iconSizes.sizeM,
    width: iconSizes.sizeM
  }
});
export default CompletionIcon;
