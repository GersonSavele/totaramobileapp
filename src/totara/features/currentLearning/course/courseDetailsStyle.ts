/**
 * This file is part of Totara Enterprise.
 *
 * Copyright (C) 2020 onwards Totara Learning Solutions LTD
 *
 * Totara Enterprise is provided only to Totara Learning Solutions
 * LTDâ€™s customers and partners, pursuant to the terms and
 * conditions of a separate agreement with Totara Learning
 * Solutions LTD or its affiliate.
 *
 * If you do not have an agreement with Totara Learning Solutions
 * LTD, you may not access, use, modify, or distribute this software.
 * Please contact [sales@totaralearning.com] for more information.
 */

import { StyleSheet } from "react-native";
import { spacedFlexRow } from "@totara/lib/styles/base";
import { TotaraTheme } from "@totara/theme/Theme";
import { margins } from "@totara/theme/constants";
import { Images } from "@resources/images";
const { colorAccent, colorNeutral7, colorSuccess, colorNeutral6, colorAlert } = TotaraTheme;

const courseDetailsStyle = StyleSheet.create({
  expandContentWrap: {
    margin: margins.marginL,
    ...spacedFlexRow
  },
  expandTextWrap: {
    ...TotaraTheme.textMedium
  },
  scrollView: {
    flex: 1,
    backgroundColor: TotaraTheme.colorNeutral2
  }
});

const completionStates = {
  notAvailable: {
    icon: "lock",
    backgroundColor: colorAccent,
    iconColor: colorNeutral7,
    borderColor: colorNeutral7
  },
  completed: {
    icon: "check",
    backgroundColor: colorSuccess,
    iconColor: colorAccent,
    borderColor: colorSuccess
  },
  autoIncomplete: {
    icon: Images.autoCompleteTick,
    backgroundColor: colorAccent,
    iconColor: colorNeutral6,
    borderColor: colorNeutral6,
    fontAwesomeIcon: false
  },
  completeFail: {
    icon: "times",
    backgroundColor: colorAlert,
    iconColor: colorAccent,
    borderColor: colorAlert
  },
  manualIncomplete: {
    backgroundColor: colorAccent,
    iconColor: colorNeutral6,
    borderColor: colorNeutral6
  }
};

export default courseDetailsStyle;
export { completionStates };
