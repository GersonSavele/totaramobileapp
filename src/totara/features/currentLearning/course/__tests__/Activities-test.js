/**
 * This file is part of Totara Enterprise.
 *
 * Copyright (C) 2019 onwards Totara Learning Solutions LTD
 *
 * Totara Enterprise is provided only to Totara Learning Solutions
 * LTDâ€™s customers and partners, pursuant to the terms and
 * conditions of a separate agreement with Totara Learning
 * Solutions LTD or its affiliate.
 *
 * If you do not have an agreement with Totara Learning Solutions
 * LTD, you may not access, use, modify, or distribute this software.
 * Please contact [sales@totaralearning.com] for more information.
 */

import renderer from "react-test-renderer";
import React from "react";
import CompletionIcon from "../CompletionIcon";
import { completionStates } from "../courseDetailsStyle";

const theme = jest.fn();

describe("Activity-Icon, testing UI activity checkbox for self-completion or auto-completion", () => {
  it("Test result : Activity is available and completion, completionStatus undefined", () => {
    const jestGetCompletionStatus = jest.fn(() => ({
      stateObj: completionStates.notAvailable,
      accessibility: undefined
    }));
    const component = renderer.create(
      <CompletionIcon
        theme={theme}
        completion="tracking_none"
        completionStatus="unknown"
        available={false}
        onGetCompletionStatus={jestGetCompletionStatus}></CompletionIcon>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("Test result : Activity is not available and completion, completionStatus undefined", () => {
    const jestGetCompletionStatus = jest.fn(() => ({
      stateObj: completionStates.notAvailable,
      accessibility: undefined
    }));
    const component = renderer.create(
      <CompletionIcon
        theme={theme}
        completion="tracking_none"
        completionStatus="unknown"
        available={true}
        onGetCompletionStatus={jestGetCompletionStatus}></CompletionIcon>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("Test result : Activity is available and auto-completion, completionStatus complete, complete-pass", () => {
    const jestGetCompletionStatus = jest.fn(() => ({ stateObj: completionStates.completed, accessibility: undefined }));
    const component = renderer.create(
      <CompletionIcon
        theme={theme}
        completion="tracking_automatic"
        completionStatus="complete"
        available={true}
        onGetCompletionStatus={jestGetCompletionStatus}></CompletionIcon>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("Test result : Activity is available and auto-completion, completionStatus incomplete", () => {
    const jestGetCompletionStatus = jest.fn(() => ({
      stateObj: completionStates.autoIncomplete,
      accessibility: undefined
    }));
    const component = renderer.create(
      <CompletionIcon
        theme={theme}
        completion="tracking_automatic"
        completionStatus="incomplete"
        available={true}
        onGetCompletionStatus={jestGetCompletionStatus}></CompletionIcon>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("Test result : Activity is available and  completionStatus completeFail", () => {
    const jestGetCompletionStatus = jest.fn(() => ({
      stateObj: completionStates.completeFail,
      accessibility: undefined
    }));

    const component = renderer.create(
      <CompletionIcon
        theme={theme}
        completion="tracking_none"
        completionStatus="completeFail"
        available={true}
        onGetCompletionStatus={jestGetCompletionStatus}></CompletionIcon>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("Test result : Activity is available and manual-completion, completionStatus incomplete", () => {
    const jestGetCompletionStatus = jest.fn(() => ({
      stateObj: completionStates.manualIncomplete,
      accessibility: undefined
    }));
    const component = renderer.create(
      <CompletionIcon
        theme={theme}
        completion="tracking_manual"
        completionStatus="incomplete"
        available={true}
        onGetCompletionStatus={jestGetCompletionStatus}></CompletionIcon>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("Test result : Activity is available and manual-completion, completionStatus complete", () => {
    const jestGetCompletionStatus = jest.fn(() => ({
      stateObj: completionStates.completed,
      accessibility: undefined
    }));
    const component = renderer.create(
      <CompletionIcon
        theme={theme}
        completion="tracking_manual"
        completionStatus="complete"
        available={true}
        onGetCompletionStatus={jestGetCompletionStatus}></CompletionIcon>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("Test result : Activity is available and manual-completion is undefined, completionStatus is undefined", () => {
    const jestGetCompletionStatus = jest.fn(() => undefined);
    const component = renderer.create(
      <CompletionIcon
        theme={theme}
        completion="tracking_none"
        completionStatus="unknown"
        available={true}
        onGetCompletionStatus={jestGetCompletionStatus}></CompletionIcon>
    );
    expect(component).toMatchSnapshot();
  });
});
