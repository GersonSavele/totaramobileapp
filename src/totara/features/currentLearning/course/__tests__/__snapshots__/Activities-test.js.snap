// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Activity-Icon, testing UI activity checkbox for self-completion or auto-completion Test result : Activity is available and  completionStatus completeFail 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginRight": 16,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 12,
          "borderStyle": "solid",
          "borderWidth": 1,
          "height": 24,
          "justifyContent": "center",
          "width": 24,
        },
        Object {
          "backgroundColor": "#953539",
          "borderColor": "#953539",
        },
      ]
    }
  >
    <Text>
      mock icon
    </Text>
  </View>
</View>
`;

exports[`Activity-Icon, testing UI activity checkbox for self-completion or auto-completion Test result : Activity is available and auto-completion, completionStatus complete, complete-pass 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginRight": 16,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 12,
          "borderStyle": "solid",
          "borderWidth": 1,
          "height": 24,
          "justifyContent": "center",
          "width": 24,
        },
        Object {
          "backgroundColor": "#69bd45",
          "borderColor": "#69bd45",
        },
      ]
    }
  >
    <Text>
      mock icon
    </Text>
  </View>
</View>
`;

exports[`Activity-Icon, testing UI activity checkbox for self-completion or auto-completion Test result : Activity is available and auto-completion, completionStatus incomplete 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginRight": 16,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 12,
          "borderStyle": "solid",
          "borderWidth": 1,
          "height": 24,
          "justifyContent": "center",
          "width": 24,
        },
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#7D7D7D",
        },
      ]
    }
  >
    <Image
      source="test-file-stub"
    />
  </View>
</View>
`;

exports[`Activity-Icon, testing UI activity checkbox for self-completion or auto-completion Test result : Activity is available and completion, completionStatus undefined 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginRight": 16,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 12,
          "borderStyle": "solid",
          "borderWidth": 1,
          "height": 24,
          "justifyContent": "center",
          "width": 24,
        },
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#4A4A4A",
        },
      ]
    }
  >
    <Text>
      mock icon
    </Text>
  </View>
</View>
`;

exports[`Activity-Icon, testing UI activity checkbox for self-completion or auto-completion Test result : Activity is available and manual-completion is undefined, completionStatus is undefined 1`] = `null`;

exports[`Activity-Icon, testing UI activity checkbox for self-completion or auto-completion Test result : Activity is available and manual-completion, completionStatus complete 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginRight": 16,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 12,
          "borderStyle": "solid",
          "borderWidth": 1,
          "height": 24,
          "justifyContent": "center",
          "width": 24,
        },
        Object {
          "backgroundColor": "#69bd45",
          "borderColor": "#69bd45",
        },
      ]
    }
  >
    <Text>
      mock icon
    </Text>
  </View>
</View>
`;

exports[`Activity-Icon, testing UI activity checkbox for self-completion or auto-completion Test result : Activity is available and manual-completion, completionStatus incomplete 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginRight": 16,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 12,
          "borderStyle": "solid",
          "borderWidth": 1,
          "height": 24,
          "justifyContent": "center",
          "width": 24,
        },
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#7D7D7D",
        },
      ]
    }
  />
</View>
`;

exports[`Activity-Icon, testing UI activity checkbox for self-completion or auto-completion Test result : Activity is not available and completion, completionStatus undefined 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginRight": 16,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 12,
          "borderStyle": "solid",
          "borderWidth": 1,
          "height": 24,
          "justifyContent": "center",
          "width": 24,
        },
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#4A4A4A",
        },
      ]
    }
  >
    <Text>
      mock icon
    </Text>
  </View>
</View>
`;
